packages/react-form/src/types.ts

```diff
@@ -99,7 +99,9 @@ export interface Form<T extends FieldBag> {
   submitErrors: FormError[];
   validate(): FormError[];
   reset(): void;
-  submit(event?: React.FormEvent): void;
+  submit(
+    event?: React.FormEvent<Element> | undefined,
```

Comment:
Do we really need both instances of `undefined` here? I believe the `event?` already implies that, while the one in the return type doesn't really add much more than `void` already does, but let me know if you think otherwise.

Reply:
Good spot on the `event?` meaning `undefined`, I've removed that one.

I think the return type still needs `undefined` though. If you look at the function it has two return statements with no values ([here](https://github.com/Shopify/quilt/blob/main/packages/react-form/src/hooks/submit.ts#L53) and [here](https://github.com/Shopify/quilt/blob/main/packages/react-form/src/hooks/submit.ts#L60)) which will return `undefined`, but if neither of those `if` statements are entered there will be no return, so `void`.

Reply:
I don't think we need `undefined` here either.

If you let TypeScript infer the type as you can see in this simple example it will actually infer it as `void` because it was not explicitly `return undefined;` but was just `return;`
<img width="363" alt="Screen Shot 2022-10-24 at 4 41 36 PM" src="https://user-images.githubusercontent.com/1389310/197625410-96c4a62f-71b3-4628-911f-ca555931df02.png">

If you slap an `async` in front of the function TypeScript then infer it to `Promise<void>`
<img width="490" alt="Screen Shot 2022-10-24 at 4 44 50 PM" src="https://user-images.githubusercontent.com/1389310/197625849-ce15631b-b8bc-461b-9ddc-84a13db356c9.png">

So in this case I would just update it to
```ts
submit(event?: React.FormEvent): Promise<void>;
```

Reply:
Plus one to this being `Promise<void>`. I also note that the default value of the generic you pass into React.FormEvent is `Element` so being explicit here serves no purpose, It is fine to keep `React.FormEvent` instead changing to `React.FormEvent<Element>` as they are functionally identical.

You mentioned in the PR description that this to match the [definition of submit](https://github.com/Shopify/quilt/blob/main/packages/react-form/src/hooks/submit.ts#L42). I note that hovering over `submit` gives the type inference of submit as `(event?: React.FormEvent) => Promise<void>`


<img width="868" alt="Screen Shot 2022-10-24 at 14 24 00" src="https://user-images.githubusercontent.com/227292/197632443-62abe71d-279c-48f2-af2e-63ca825d7a74.png">

Reply:
Thanks for the great feedback @egjiri & @BPScott ... 👍 

I've updated it to `submit(event?: React.FormEvent): Promise<void>;`.